roles/rhel_subscribe/tasks/main.yml

```diff
@@ -33,36 +39,58 @@
   register: result
   until: result | success
 
-- name: RedHat subscriptions
+- name: Is host already registered?
+  command: subscription-manager version
+  register: rh_subscribed
+  changed_when: "'not registered' in rh_subscribed.stdout"
+  ignore_errors: yes
```

Comment:
`failed_when: false` so that we don't affect playbook return code

Reply:
Ack